app/recipe/views.py

```diff
@@ -0,0 +1,80 @@
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework import viewsets, mixins, status
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+
+from core.models import Tag, Ingredient, Recipe
+from recipe import serializers
+
+
+class BaseRecipeAttrViewSet(viewsets.GenericViewSet,
+                            mixins.ListModelMixin,
+                            mixins.CreateModelMixin):
+    """Base generic viewset"""
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        assigned_only = bool(
```

Comment:
Why do you do a double cast here?

Reply:
Because query params come always as a string so it will be like `bool('0')` or `bool('1')` which always returns True, so casting first to integer solves this issue.